Backend/Router/netflix.js

```diff
@@ -0,0 +1,121 @@
+import express from "express";
+
+import data from "../data/netflix-titles.json";
+
+// ----------------- GLOBAL VARIABLES ---------------- //
+const router = express.Router();
+
+// --------------------- FUNCTIONS ------------------- //
+
+// slice depending on page query (10 results per page)
+const sliceData = (array, num, res) => {
+  array = array.slice(num * 10, (num + 1) * 10);
+  return array.length < 1
+    ? res.status(404).send({ error: "Sorry, no more pages" })
+    : array;
+};
+
+// for adding queries
+const handleQueries = (req, res, arr) => {
+  let filteredArr;
+  const { year, genre, page, country } = req.query;
+  const queryArr = [
+    { name: "", value: "" },
+    { name: "release_year", value: year },
+    { name: "listed_in", value: genre },
+    { name: "country", value: country }
+  ];
+  let i = 0;
+
+  // if a query exists the array will be filtered after that query.
+  if (year || genre || country) {
+    // eslint-disable-next-line array-callback-return
+    filteredArr = arr.filter((item) => {
+      // eslint-disable-next-line no-unused-expressions
+      i < 3 ? (i += 1) : (i = 1); // used to loop through array with queries
+      if (queryArr[i].value) {
+        return item[queryArr[i].name].toString().toLowerCase().replace(" ", "")
+          .includes(queryArr[i].value);
+      }
+    });
+  } else {
+    filteredArr = arr;
+  }
+
+  if (page) {
+    // user can see more results with query page
+    filteredArr = sliceData(filteredArr, Number(page), res);
+  }
+
+  // if no data is returned from filtering (no match with params/query) a message says so
+  if (filteredArr.length === 0) {
+    res.status(404).send({ error: "Sorry, no titles found for this search" });
```

Comment:
you use this structure a lot, maybe it is a good idea to extract it to a separate function and call it if needed? 🤓